packages/cli/src/databases/db-connection-options.ts

```diff
@@ -0,0 +1,146 @@
+import { GlobalConfig } from '@n8n/config';
+import { entities } from '@n8n/db';
+import { Service } from '@n8n/di';
+import type { DataSourceOptions, LoggerOptions } from '@n8n/typeorm';
+import type { MysqlConnectionOptions } from '@n8n/typeorm/driver/mysql/MysqlConnectionOptions';
+import type { PostgresConnectionOptions } from '@n8n/typeorm/driver/postgres/PostgresConnectionOptions';
+import type { SqliteConnectionOptions } from '@n8n/typeorm/driver/sqlite/SqliteConnectionOptions';
+import type { SqlitePooledConnectionOptions } from '@n8n/typeorm/driver/sqlite-pooled/SqlitePooledConnectionOptions';
+import { InstanceSettings } from 'n8n-core';
+import { UserError } from 'n8n-workflow';
+import path from 'path';
+import type { TlsOptions } from 'tls';
+
+import { InsightsByPeriod } from '@/modules/insights/database/entities/insights-by-period';
+import { InsightsMetadata } from '@/modules/insights/database/entities/insights-metadata';
+import { InsightsRaw } from '@/modules/insights/database/entities/insights-raw';
+
+import { mysqlMigrations } from './migrations/mysqldb';
+import { postgresMigrations } from './migrations/postgresdb';
+import { sqliteMigrations } from './migrations/sqlite';
+import { subscribers } from './subscribers';
+
+@Service()
+export class DBConnectionOptions {
+	constructor(
+		private readonly globalConfig: GlobalConfig,
+		private readonly instanceSettings: InstanceSettings,
```

Comment:
When this moves to `@n8n/db` we won't have access to `InstanceSettings`. Can we pass in `n8nFolder` only?

Reply:
moved `InstanceSettingsConfig` to `@n8n/config`, and switched to that instead.